### ISSUE TITLE
Is this message generated from your custom component? '[glocaltokens.client] [!] Could not get access token.'

### ISSUE THREAD

**Issue by deepcoder** [leikoilja/ha-google-home#93]
Your custom component had been running fine for a couple days. Then this morning, my home assistant log showed this error and then the error and warning shown below started filling the log every second until I shut down the server. Are these messages from your custom component?

```
2021-03-15 02:07:36 ERROR (MainThread) [custom_components.google_home] Timeout fetching sensor data

2021-03-15 02:08:16 ERROR (SyncWorker_8) [glocaltokens.client] [!] Could not get access token.
2021-03-15 02:08:16 WARNING (SyncWorker_8) [glocaltokens.client] The access token has expired. Getting a new one.
2021-03-15 02:08:16 ERROR (SyncWorker_8) [glocaltokens.client] [!] Could not get access token.
2021-03-15 02:08:16 WARNING (SyncWorker_8) [glocaltokens.client] The access token has expired. Getting a new one.
2021-03-15 02:08:16 ERROR (SyncWorker_8) [glocaltokens.client] [!] Could not get access token.
```

**Comment by ArnyminerZ** [comment]
Are you sure that you are using the last version? This was a bug found some versions ago

**Comment by leikoilja** [comment]
@ArnyminerZ, is right. This should have been fixed. @deepcoder can you please try reinstalling the component. I suggest removing HACS repo entirely, making sure the `custom_component/google_home` directory is deleted and then adding a custom repo in HACS once again selecting the latest `1.0.0` release. :) 
Please keep us posted

**Comment by deepcoder** [comment]
I upgraded to the version that was current as of Mar 16, 2021 and it blew up again.

I'm not a HAC's guy, I installed via your 'Copy method' installation steps.

with the second error below, the HA log gets inundated errors, 3 to 4 per SECOND. Really need a way to tone that down!!!!

first with this error about 15 hours after starting HA, second about 60 hours after start.

```
2021-03-16 10:45:39 INFO (MainThread) [homeassistant.core] Starting Home Assistant
```
...
...
...

```
2021-03-17 03:33:37 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 126, in _handle_refresh_interval
    await self.async_refresh()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 200, in async_refresh
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 260, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 295, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 319, in _async_write_ha_state
    sstate = self.state
  File "/config/custom_components/google_home/sensor.py", line 141, in state
    alarms = self._get_alarms_data()
  File "/config/custom_components/google_home/sensor.py", line 156, in _get_alarms_data
    return self.as_dict(device.get_sorted_alarms())
AttributeError: 'NoneType' object has no attribute 'get_sorted_alarms'
```

then with this error about 60 hours later, thousands of these in log until I removed the integration.


```
2021-03-18 02:22:09 ERROR (SyncWorker_15) [glocaltokens.client] [!] Could not get access token.
2021-03-18 02:22:09 WARNING (SyncWorker_15) [glocaltokens.client] The access token has expired. Getting a new one.
2021-03-18 02:22:09 ERROR (SyncWorker_15) [glocaltokens.client] [!] Could not get access token.
2021-03-18 02:22:09 WARNING (SyncWorker_15) [glocaltokens.client] The access token has expired. Getting a new one.

```

**Comment by DurgNomis-drol** [comment]
The first problem I also ran into yesterday, wasn't sure why i did but now i can see that i am not the only one. But it should be split up into a separate issue. 😄 

The second one is strange if you are using the latest version. Can you confirm the version of `glocaltokens` by running `pip show glocaltokens`? I have personally expired that HA doesn't always update underlay packages unfortunately. Though i think this is very rare.

**Comment by deepcoder** [comment]
from within docker container via portainer:

Home Assistant 2021.3.3 container from Home Assistant docker hub:
```
bash-5.0# pip show glocaltokens
Name: glocaltokens
Version: 0.2.4
Summary: Tool to extract Google device local authentication tokens in Python
Home-page: https://github.com/leikoilja/glocaltokens
Author: Ilja Leiko
Author-email: leikoilja@gmail.com
License: MIT
Location: /usr/local/lib/python3.8/site-packages
Requires: zeroconf, grpcio-tools, simplejson, grpcio, requests, gpsoauth
Required-by: 
bash-5.0# 

```

**Comment by DurgNomis-drol** [comment]
That is the correct version. This seems very strange... 

Some more questions:
- Are you running Home Assistant in network `Host` mode?
- Are you using an app password or your normal password?
- Are you using an `@gmail.com` or a custom domain?

Next step would be for you to test if `glocaltokens` package works for you so we can try to isolate the issue.

**Comment by bacco007** [comment]
I'm using HA OS, and its not installing the python library `glocaltokens` - which appears to be failing on the install of `grpcio`

>  FileNotFoundError: [Errno 2] No such file or directory: 'c++'

**Comment by DurgNomis-drol** [comment]
@bacco007 Please open another issue as this is not related to this issue 😊

**Comment by leikoilja** [comment]
@deepcoder, there is now a new release of the integration. 
Please upgrade to `1.0.1` using HACS.
Also make sure you have the logger enabled in your `configuration.yaml`:
```yaml
logger:
  default: info
  logs:
    custom_components.google_home: debug
```
That will now print much more elaborate logs. Please post them here (**IMPORTANT** make sure not to paste your sensitive info here as `master_token` or `login`). These elaborate logs (paste as code block) will help us get a better understanding why you are getting an issue

**Comment by bacco007** [comment]
I've confirmed that my username and app password are correct (i've also tried another google account i've got that doesnt have 2FA enabled, same error)

```
2021-03-26 17:42:23 DEBUG (MainThread) [glocaltokens.client] Initializing new GLocalAuthenticationTokens instance.
2021-03-26 17:42:23 DEBUG (MainThread) [glocaltokens.client] Set self username to "b*****************", password to "v***************", master_token to "" and android_id to 
2021-03-26 17:42:23 DEBUG (MainThread) [glocaltokens.client] Set self access_token, homegraph, access_token_date and homegraph_date to None
2021-03-26 17:42:23 DEBUG (SyncWorker_19) [glocaltokens.client] There is not any stored master_token, logging in...
2021-03-26 17:42:23 DEBUG (SyncWorker_19) [glocaltokens.client] There is not any stored android_id, generating a new one
2021-03-26 17:42:23 DEBUG (SyncWorker_19) [glocaltokens.client] Generating mac...
2021-03-26 17:42:23 DEBUG (SyncWorker_19) [glocaltokens.client] Generated mac: E108D5F5417647
2021-03-26 17:42:23 ERROR (SyncWorker_19) [glocaltokens.client] [!] Could not get master token.
2021-03-26 17:42:23 DEBUG (SyncWorker_19) [glocaltokens.client] Request response: {'Error': 'BadAuthentication'}
2021-03-26 17:42:23 ERROR (MainThread) [custom_components.google_home] 
```

**Comment by leikoilja** [comment]
@bacco007 & guys, i actually think it has nothing to do with google accounts themself but with the dependency libraries used. We have seen `BadAuthentication` before on some versions of `requests` library that are used under the hood. It might be that some other integrations or packages that you use in parallel with `Google Home` are enforcing different (lower) version of the `requests` library. 
@DurgNomis-drol, I know you are more expert in that field. What was the good way to see the version of `requests` installed in the HA setup? 🤔  😅

**Comment by leikoilja** [comment]
The underlying issue now looks very much like existing ticket #95. 
I am going to close this one and move the conversation there. 
@deepcoder if you still encounter ANY kind of different issue then `BadAuthentication` after updating to the latest available release please open a new issue specifically about the problem you would encounter :)

**Comment by DurgNomis-drol** [comment]
You can check which version you have installed by running `pIp show package_name` in the command line in the docker container of HA.

The important packages is `requests`,`urllib3`,`chardet` and `gpsoauth`. The problem is caused by a bug on the Google servers in how urllib3 have implemented SNI support. It should have been solved, but as @leikoilja states, if some other things are forcing a wrong version, this bug will show up.
